Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/ConceptIDLookup.java

```diff
@@ -95,4 +107,21 @@
         
         return new ResponseEntity<String>(msg.getXML(xmlEntries), HttpStatus.OK);
     }
+
+    private void addAlternativeIds(String id, ConceptEntry entry)
+            throws IllegalAccessException, LuceneException, IndexerRunningException {
+
+        // Check if this id is same as the id sent by user
+        if (!entry.getId().equalsIgnoreCase(id)) {
+            entry.getAlternativeIds().add(id);
+        }
+
+        // Fetch the internal concept id based on the wordnetid
+        Map<String, String> fieldMap = new HashMap<String, String>();
+        fieldMap.put(SearchFieldNames.WORDNETID, entry.getWordnetId());
+        ConceptEntry[] conceptEntries = indexService.searchForConcepts(fieldMap, "AND");
```

Comment:
there should only be one concept with a specific concept id and if there is a concept wrapper for a wordnet concept, there is not extra wordnet concept indexed. In other words, the id you get in line 124 should be the same as the one you already have?